src/main/java/org/sagebionetworks/bridge/hibernate/HibernateSurveyDao.java

```diff
@@ -0,0 +1,365 @@
+    
+    private static final String GET_ACTIVE_PUBLISH = "FROM HibernateSurvey as survey " + 
+            "WHERE studyKey = :studyKey AND guid = :guid AND deleted = 0 AND published = :published ORDER BY createdOn DESC";
+    
+    private static final String GET_ACTIVE_PUBLISH_WITH_IDENTIFIER = "FROM HibernateSurvey as survey " + 
+            "WHERE studyKey = :studyKey AND identifier = :identifier AND deleted = 0 AND published = :published ORDER BY createdOn DESC";
+    
+    private static final String GET_ACTIVE_WITH_IDENTIFIER = "FROM HibernateSurvey as survey " + 
+            "WHERE studyKey = :studyKey AND identifier = :identifier AND createdOn = :createdOn AND deleted = 0 ORDER BY createdOn DESC";
+    
+    private static final String GET_ELEMENTS = "FROM HibernateSurveyElement as surveyElement " + 
+            "WHERE surveyGuid = :surveyGuid AND createdOn = :createdOn ORDER BY createdOn DESC";
+    
+    private static final String DELETE_SURVEY_ELEMENTS = "DELETE FROM HibernateSurveyElement WHERE surveyGuid = :surveyGuid AND createdOn = :createdOn";
+    
+    private HibernateHelper hibernateHelper;
+    
+    private UploadSchemaService uploadSchemaService;
+    
+    /** This makes interfacing with Hibernate easier. */
+    @Resource(name = "basicHibernateHelper")
+    public final void setHibernateHelper(HibernateHelper hibernateHelper) {
+        this.hibernateHelper = hibernateHelper;
+    }
+    
+    @Autowired
+    public final void setUploadSchemaService(UploadSchemaService uploadSchemaService) {
+        this.uploadSchemaService = uploadSchemaService;
+    }
+    
+    @Override
+    public Survey createSurvey(Survey survey) {
+        System.out.println("createSurvey");
+        checkNotNull(survey.getStudyIdentifier(), "Survey study identifier is null");
+        if (survey.getGuid() == null) {
+            survey.setGuid(generateGuid());
+        }
+        long time = DateUtils.getCurrentMillisFromEpoch();
+        survey.setCreatedOn(time);
+        survey.setModifiedOn(time);
+        survey.setSchemaRevision(null);
+        survey.setPublished(false);
+        survey.setDeleted(false);
+        survey.setVersion(null);
+        hibernateHelper.create(survey, null);
+        System.out.println("12");
+        return survey;
+    }
+
+    @Override
+    public Survey updateSurvey(StudyIdentifier studyIdentifier, Survey survey) {
+        Survey existingSurvey = getSurvey(studyIdentifier, survey, true);
+        
+        if (existingSurvey != null) {
+            hibernateHelper.update(survey, null);
+            return survey;
+        } else {
+            return createSurvey(survey);
+        }
+    }
+
+    @Override
+    public Survey versionSurvey(StudyIdentifier studyIdentifier, GuidCreatedOnVersionHolder keys) {
+        Survey existingSurvey = getSurvey(studyIdentifier, keys, false);
+        System.out.println(existingSurvey);
+        HibernateSurvey copy = new HibernateSurvey(existingSurvey);
```

Comment:
The DynamoSurveyDao implementation also cleared the published and deleted flags and the version number, and also set the createdOn and modifiedOn timestamps.